packages/store/addon/-private/system/snapshot.ts

```diff
@@ -422,14 +476,14 @@ export default class Snapshot {
     @param {Function} callback the callback to execute
     @param {Object} [binding] the value to which the callback's `this` should be bound
   */
-  eachRelationship(callback, binding) {
+  eachRelationship(callback: (key: string, meta: RelationshipSchema) => void, binding?: any): void {
     if (CUSTOM_MODEL_CLASS) {
       let relationshipDefs = this._store._relationshipsDefinitionFor(this.modelName, this.identifier);
       Object.keys(relationshipDefs).forEach(key => {
         callback.call(binding, key, relationshipDefs[key]);
       });
     } else {
-      this.record.eachRelationship(callback, binding);
+      (this.record as DSModel).eachRelationship(callback, binding);
```

Comment:
As in the previous place where we had the same check on `CUSTOM_MODEL_CLASS`, it'd be great if we could push this into the type system or at least give it a comment explaining why it's safe so folks don't break it!